components/movementsensor/imuwit/imu.go

```diff
@@ -198,45 +196,45 @@ func NewWit(
 	i.port = port
 
 	portReader := bufio.NewReader(port)
+	i.startUpdateLoop(ctx, portReader, logger)
 
-	ctx, i.cancelFunc = context.WithCancel(context.Background())
-	i.activeBackgroundWorkers.Add(1)
+	return &i, nil
+}
+
+func (imu *wit) startUpdateLoop(ctx context.Context, portReader *bufio.Reader, logger golog.Logger) {
+	ctx, imu.cancelFunc = context.WithCancel(context.Background())
+	imu.activeBackgroundWorkers.Add(1)
```

Comment:
[minor] What do we think of enforcing a time on this loop?

Reply:
what do you mean by time? Also, any reason not to derive from parent context?

Reply:
Not that I know of, this was just refactoring code. I can change to the parental context and test.

Reply:
Re: you r comment about time - should enforce Ticker or Timer Structure on this for loop to poll at a set rate that we are in control of? Kind of like the vectornav model's logic.

Reply:
I think it's not necessary. With vectornav we need to pull the data but with Wit it seems liked they are pushed on the wire so the ReadString function will do an efficient job.